src/Sign.Cli/AzureCredentialOptions.cs

```diff
@@ -5,26 +5,49 @@
 using System.CommandLine;
 using System.CommandLine.Invocation;
 using System.CommandLine.IO;
+using System.CommandLine.Parsing;
 using Azure.Core;
 using Azure.Identity;
 
 namespace Sign.Cli
 {
     internal sealed class AzureCredentialOptions
     {
+        internal Option<string?> CredentialTypeOption { get; } = new Option<string?>(["-act", "--azure-credential-type"], Resources.CredentialTypeOptionDescription).FromAmong(
+            AzureCredentialType.Environment);
         internal Option<bool?> ManagedIdentityOption { get; } = new(["-kvm", "--azure-key-vault-managed-identity"], Resources.ManagedIdentityOptionDescription);
         internal Option<string?> TenantIdOption { get; } = new(["-kvt", "--azure-key-vault-tenant-id"], Resources.TenantIdOptionDescription);
         internal Option<string?> ClientIdOption { get; } = new(["-kvi", "--azure-key-vault-client-id"], Resources.ClientIdOptionDescription);
         internal Option<string?> ClientSecretOption { get; } = new(["-kvs", "--azure-key-vault-client-secret"], Resources.ClientSecretOptionDescription);
 
         internal void AddOptionsToCommand(Command command)
         {
+            command.AddOption(CredentialTypeOption);
             command.AddOption(ManagedIdentityOption);
             command.AddOption(TenantIdOption);
             command.AddOption(ClientIdOption);
             command.AddOption(ClientSecretOption);
         }
 
+        internal DefaultAzureCredentialOptions CreateDefaultAzureCredentialOptions(ParseResult parseResult)
```

Comment:
This will get complicated as we add explicit support for specific authentication strategies.

I was envisioning something simpler and more scalable:

```C#
if (string.Equals(AzureCredentialType.Environment, credentialType, StringComparison.OrdinalIgnoreCase))
{
    return new EnvironmentCredential();
}
else
{
    return new DefaultAzureCredential();
}
```

Reply:
After looking at documentation, I see that [`DefaultAzureCredential`'s default constructor is only intended for mocking](https://learn.microsoft.com/dotnet/api/azure.identity.defaultazurecredential.-ctor?view=azure-dotnet#overloads).  We shouldn't be using it here.

`EnvironmentCredential` does have a default constructor that is fine to use here.

However, [`ManagedIdentityCredential`'s default constructor is also intended for mocking](https://learn.microsoft.com/dotnet/api/azure.identity.managedidentitycredential?view=azure-dotnet#constructors).

I'm not sure yet what it would look like for us to construct a `ManagedIdentityCredential` with non-default options.  (I haven't looked at what `DefaultAzureCredential` does.)

Reply:
With `new DefaultAzureCredential()` we would be using this constructor: `public DefaultAzureCredential(bool includeInteractiveCredentials = false)`. I thought that the advantage of `DefaultAzureCredential` with the options would be that we would not need to construct each of the `Credential` classes and just use the logic that is inside `DefaultAzureCredential`. Otherwise we would probably create something that looks a lot like the implementation behind `DefaultAzureCredential`. And we would need to maintain that then. With this strategy we allow a user to specify which one the credential types they would want to use. And we can still change this in the future if we really need to?